src/main/java/seedu/address/model/borrower/Borrower.java

```diff
@@ -62,7 +62,10 @@ public boolean isSameBorrower(Borrower otherBorrower) {
         }
 
         return otherBorrower != null
-                && otherBorrower.getBorrowerId().equals(getBorrowerId());
+                && otherBorrower.getBorrowerId().equals(getBorrowerId())
+                && otherBorrower.getName().equals(getName())
+                && (otherBorrower.getPhone().equals(getPhone())
+                    || otherBorrower.getEmail().equals(getEmail()));
```

Comment:
Looking at lines 35-42 in `BorrowerTest`, I suggest we only compare borrowerId for `isSameBorrower(Borrower borrower)`, and impose this strict equality in the `equals(Borrower borrower)` method instead

Reply:
I think this suggestion is pretty valid, since Borrowers may change their phone and email. Will update this!